### PR TITLE
Harvest: refactor deposit type to claim type

### PR DESCRIPTION
**Pull request by denalimarsh**
- Refactor deposit type to claim type
- Remove deposit type from `Deposit` and `Withdraw` msgs and keeper functions
- Update tests, spec

### CODE REVIEW

File: x/harvest/keeper/deposit.go [hunk 1]
```diff
@@ -9,26 +9,21 @@ import (
...
-	default:
-		return sdkerrors.Wrap(types.ErrInvalidDepositType, string(depositType))
-	}
+	err = k.supplyKeeper.SendCoinsFromAccountToModule(ctx, depositor, types.ModuleAccountName, sdk.NewCoins(amount))
 	if err != nil {
 		return err
 	}
 
-	deposit, found := k.GetDeposit(ctx, depositor, amount.Denom, depositType)
+	deposit, found := k.GetDeposit(ctx, depositor, amount.Denom)
```
Comment (karzak) [inline x/harvest/keeper/deposit.go:24]: Add an `IncrementTotalDeposited` call here

File: x/harvest/keeper/keeper.go [hunk 2]
```diff
@@ -114,11 +114,10 @@ func (k Keeper) IterateDeposits(ctx sdk.Context, cb func(deposit types.Deposit)
 	}
 }
 
-// IterateDepositsByTypeAndDenom iterates over all deposit objects in the store with the matching deposit type and deposit denom and performs a callback function
-func (k Keeper) IterateDepositsByTypeAndDenom(ctx sdk.Context, depositType types.DepositType, depositDenom string, cb func(deposit types.Deposit) (stop bool)) {
-
+// IterateDepositsByDenom iterates over all deposit objects in the store with the matching deposit denom and performs a callback function
+func (k Keeper) IterateDepositsByDenom(ctx sdk.Context, depositDenom string, cb func(deposit types.Deposit) (stop bool)) {
```
Comment (karzak) [inline x/harvest/keeper/keeper.go:118]: Add `GetTotalDeposited`, `SetTotalDeposited`, `IncrementTotalDeposited`, and `DecrementTotalDeposited` methods

File: x/harvest/keeper/deposit.go [hunk 3]
```diff
@@ -75,22 +54,16 @@ func (k Keeper) ValidateLPDeposit(ctx sdk.Context, amount sdk.Coin, depositType
...
 	}
 
-	var err error
-	switch depositType {
-	case types.LP:
-		err = k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleAccountName, depositor, sdk.NewCoins(amount))
-	default:
-		return sdkerrors.Wrap(types.ErrInvalidDepositType, string(depositType))
-	}
+	err := k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleAccountName, depositor, sdk.NewCoins(amount))
```
Comment (karzak) [inline x/harvest/keeper/deposit.go:66]: Add `DecrementTotalDeposits` after here

File: x/harvest/keeper/deposit.go [hunk 4]
```diff
@@ -125,12 +96,8 @@ func (k Keeper) Withdraw(ctx sdk.Context, depositor sdk.AccAddress, amount sdk.C
 }
 
 // GetTotalDeposited returns the total amount deposited for the input deposit type and deposit denom
-func (k Keeper) GetTotalDeposited(ctx sdk.Context, depositType types.DepositType, depositDenom string) (total sdk.Int) {
-
+func (k Keeper) GetTotalDeposited(ctx sdk.Context, depositDenom string) (total sdk.Int) {
```
Comment (karzak) [inline x/harvest/keeper/deposit.go:99]: This will have to be tracked separately from the module account once borrows are enabled.

**Review by karzak — CHANGES_REQUESTED** [review]
Code changes look correct. Before merging

1. Update decoder test in simulation to fix build
2. Add legacy types folder

Non-blocking
1. See comments about tracking total deposits in the store rather than using the module account. Once borrows are enabled, this will no longer be accurate. Fine to scope it to a separate PR/Chore though.

File: x/harvest/keeper/deposit.go (hunk 1 above)
Comment (denalimarsh) [reply x/harvest/keeper/deposit.go:24]: Created an Asana card for this

File: x/harvest/keeper/keeper.go (hunk 2 above)
Comment (denalimarsh) [reply x/harvest/keeper/keeper.go:118]: See https://github.com/Kava-Labs/kava/pull/716#discussion_r522122035

File: x/harvest/keeper/deposit.go (hunk 3 above)
Comment (denalimarsh) [reply x/harvest/keeper/deposit.go:66]: See https://github.com/Kava-Labs/kava/pull/716#discussion_r522122035

File: x/harvest/keeper/deposit.go (hunk 4 above)
Comment (denalimarsh) [reply x/harvest/keeper/deposit.go:99]: See https://github.com/Kava-Labs/kava/pull/716#discussion_r522122035

File: x/harvest/types/claim.go [hunk 5]
```diff
@@ -9,15 +9,15 @@ type Claim struct {
 	Owner        sdk.AccAddress `json:"owner" yaml:"owner"`
 	DepositDenom string         `json:"deposit_denom" yaml:"deposit_denom"`
 	Amount       sdk.Coin       `json:"amount" yaml:"amount"`
-	Type         DepositType    `json:"type" yaml:"type"`
+	Type         ClaimType      `json:"claim_type" yaml:"claim_type"`
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/types/claim.go:12]: Added in https://github.com/Kava-Labs/kava/pull/716/commits/c2a7c987717a511910b6043eeb5438c8d4b596cb. Please double check contents as it's my first legacy types file.

**Review by karzak — APPROVED** [review]
LGTM, just update `simulation/decoder_test.go:30:29` to fix build before merging

**Pull request merged**